pkg/collectors/ptp_collector.go

```diff
@@ -0,0 +1,186 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+package collectors
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/redhat-partner-solutions/vse-sync-testsuite/pkg/callbacks"
+	"github.com/redhat-partner-solutions/vse-sync-testsuite/pkg/clients"
+	"github.com/redhat-partner-solutions/vse-sync-testsuite/pkg/collectors/devices"
+)
+
+type PTPCollector struct {
+	lastPoll        time.Time
+	callback        callbacks.Callback
+	data            map[string]interface{}
+	running         map[string]bool
+	DataTypes       [3]string
+	ctx             clients.ContainerContext
+	interfaceName   string
+	inversePollRate float64
+}
+
+const (
+	VendorIntel = "0x8086"
+	DeviceE810  = "0x1593"
+
+	DeivceInfo = "device-info"
+	DLLInfo    = "dll-info"
+	GNNSSTTY   = "gnss-tty"
+	All        = "all"
+
+	PTPNamespace  = "openshift-ptp"
+	PodNamePrefix = "linuxptp-daemon-"
+	PTPContainer  = "linuxptp-daemon-container"
+)
+
+var collectables = [3]string{
+	DeivceInfo,
+	DLLInfo,
+	GNNSSTTY,
+}
+
+func (ptpDev *PTPCollector) getNotCollectableError(key string) error {
+	return fmt.Errorf("key %s is not a colletable of %T", key, ptpDev)
+}
+
+func (ptpDev *PTPCollector) getErrorIfNotCollectable(key string) error {
+	if _, ok := ptpDev.data[key]; !ok {
+		return ptpDev.getNotCollectableError(key)
+	} else {
+		return nil
+	}
+}
+
+func (ptpDev *PTPCollector) Start(key string) error {
+	switch key {
+	case All:
+		for _, dataType := range ptpDev.DataTypes[:] {
+			log.Debugf("starting: %s", dataType)
+			ptpDev.running[dataType] = true
+		}
+	default:
+		err := ptpDev.getErrorIfNotCollectable(key)
+		if err != nil {
+			return err
+		}
+		ptpDev.running[key] = true
+	}
+	return nil
+}
+
+// Checks to see if the enou
```

Comment:
What is it looking for! The suspense is too much.

This line can probably be removed.